Fix for MIFOSX-2782

gkrishnan724 opened the pull request:
Issue:
In Admin >> Organisation >> Payment Type >> Add Payment type , Position field was not marked mandatory(*) and the input type was number
Fix:
Marked field mandatory and gave appropriate error message on invalid input 
![screenshot from 2016-11-29 19-03-21](https://cloud.githubusercontent.com/assets/21004514/20711890/d8d175de-b667-11e6-9e6b-4e405b72d3df.png)

nazeer1100126 commented:
If position field is mandatory, platform should return validation saying position field is mandatory. You should not hard code in js and should not modify the api validation directive.

gkrishnan724 commented:
@nazeer1100126  Hi,  I changed Api-validation directive since it did not show in which field there was an error, I added the error.field which would show the corresponding field in which the error is present (This works for all forms) the "-Position" is due to my addition

Regarding the code in JS , I assume the file notificationwidget.js is used to fetch the errors from the server and push it into an array. So the work of api-validation directive is to display the error messages from the given array. However, In this case when an invalid input is given to the position field, the server does not return any error messages(it returns an empty array) thus the api-validation fails here. Hence , I did the validation in JS and pushed the error messages into the array and api-validation displays them.
